wheatley/bot.py

```diff
@@ -112,10 +112,17 @@ def _on_row_gen_change(self, row_gen_json):
             self.logger.warning(e)
 
     def _on_size_change(self):
-        if not self.row_generator.is_tower_size_valid(self._tower.number_of_bells):
-            self.logger.warning(f"Row generation requires {self.row_generator.number_of_bells} \
-bells, but the current tower has {self._tower.number_of_bells}.  Wheatley will crash when you go \
-into changes unless something is done!")
+        if self._tower.number_of_bells < self.row_generator.stage:
+            self.logger.warning(f"Row generation requires at least {self.row_generator.stage} bells, "
+                                + f"but the current tower has {self._tower.number_of_bells}. "
+                                + "Bells will be omitted!")
```

Comment:
It's a curious effect with omitted bells, e.g. try Plain Bob Minor on 4 bells. It will only ring 1-4 when they are in the first four places in a row, otherwise leaving a gap for 5/6 and ignoring the last two bells in a row.
I'm happy to keep it or revert back to throwing on the assert at start of method

Reply:
Another interesting thing to try is changing tower size while ringing. Wheatley now carries on like a champ, though the rhythm gets a bit confused.

Reply:
I think that tripping an assert at the start of method if the row gen wants more bells than the tower has is probably the best strategy (provided there's some kind of warning that the assert will trip, as there currently is).  Much as I enjoy funky behaviour, it's probably best to just crash gracefully than to behave in an undefined way that might make people think that Wheatley is broken.

I doubt a touch rung by any human ringers would survive the tower size changing half way through - it's nice to know that even if someone _did_ manage to do so, Wheatley will mostly behave OK.

Reply:
I'll have a look at just getting Wheatley to refuse to ring and emit this message again as an error if the number of bells is too small

Reply:
Noice :+1: